### PR TITLE
Insert Go Sample's Prerequisites

### PR DESCRIPTION
**Pull request by pongad**
This change adds to Go Apiary samples instructions users can follow
to enable cloud services and set up authentication for them.

This change also adds version renaming for Go to cope with naming
inconsistency in clouduseraccounts/beta.

Finally this change also fixes unchecked warnings.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -18,6 +18,14 @@ buildscript {
   }
 }
 
+allprojects {
+  gradle.projectsEvaluated {
```
Comment (vchudnov-g) [inline build.gradle]: In general it's better to have unrelated changes in separate CLs.

File: vgen/src/main/java/io/gapi/vgen/go/GoDiscoveryContext.java [hunk 2]
```diff
@@ -130,4 +131,14 @@ public boolean isPageStreaming(Method method) {
     }
     return hasPageToken && hasNextPageToken;
   }
+
+  private static final ImmutableTable<String, String, String> API_VERSION_RENAME =
```
Comment (vchudnov-g) [inline vgen/src/main/java/io/gapi/vgen/go/GoDiscoveryContext.java]: Add a comment saying that we need this because in some cases there is a mismatch between X and Y version numbers. Also mention that it's a table of API name and version (as found in X)  to the renamed version as found in Y.

Add a TODO(pongad) to find a more sustainable solution to this.

File: vgen/src/main/resources/io/gapi/vgen/go/discovery_fragment.snip [hunk 3]
```diff
@@ -9,13 +9,23 @@
 @snippet generateClass(method, body)
 package main
 
+// PRE-REQUISITES:
+// ---------------
+// 1. If not already done, enable the {@context.getService.getTitle} and check quota for your project at
```
Comment (vchudnov-g) [inline vgen/src/main/resources/io/gapi/vgen/go/discovery_fragment.snip]: s/quota/the quota/

File: build.gradle (hunk 1 above)
Comment (pongad) [inline build.gradle]: I thought I would include it because the change is so small. I see your point though, I'll back it out.

File: vgen/src/main/java/io/gapi/vgen/go/GoDiscoveryContext.java (hunk 2 above)
Comment (pongad) [inline vgen/src/main/java/io/gapi/vgen/go/GoDiscoveryContext.java]: Done.

File: vgen/src/main/resources/io/gapi/vgen/go/discovery_fragment.snip (hunk 3 above)
Comment (pongad) [inline vgen/src/main/resources/io/gapi/vgen/go/discovery_fragment.snip]: Done.

File: vgen/src/main/java/io/gapi/vgen/go/GoDiscoveryContext.java [hunk 4]
```diff
@@ -130,4 +131,20 @@ public boolean isPageStreaming(Method method) {
...
+      ImmutableTable.<String, String, String>builder()
+          .put("clouduseraccounts", "beta", "v0.beta")
+          .build();
+
+  // TODO(pongad): Find a more sustainable solution to this.
```
Comment (vchudnov-g) [inline vgen/src/main/java/io/gapi/vgen/go/GoDiscoveryContext.java]: It reads funny to have this before the main comment. Maybe move it at the end of the javadoc?

**Comment by vchudnov-g** [comment]
Just one minor cosmetic suggestion, but LGTM.

File: vgen/src/main/java/io/gapi/vgen/go/GoDiscoveryContext.java (hunk 4 above)
Comment (pongad) [inline vgen/src/main/java/io/gapi/vgen/go/GoDiscoveryContext.java]: Done.
